### PR TITLE
[FEATURE] Adding Restart / Stop Tests for admin and user

### PR DESCRIPTION
**Pull request by satyammittal**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [ ] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [x] I am an active contributor to the project.

---

1. Added restart and stop button for admin for any test.
2. Users can restart and stop their tests in customized tests panel.

### CODE REVIEW

File: templates/test/list_table.html [hunk 1]
```diff
@@ -19,7 +19,7 @@
                         <td data-title="Run Errors">{{ "Unknown" if not test.finished else ("Yes" if test.failed else "No") }}</td>
                         <td data-title="Completed">
                             <a href="{{ url_for('test.by_id', test_id=test.id) }}">{{ "Yes" if test.finished else "No" }}</a>
-                            {% if user.is_admin or "{{ url_for(request.endpoint) }}" == "{{ url_for('custom.index'}}" %}
+                            {% if user.is_admin or url_for(request.endpoint) == url_for('custom.index') %}
```
Comment (canihavesomecoffee) [inline templates/test/list_table.html]: I'm still wondering why you compare using url_for when you can just compare the endpoints?

File: templates/test/list_table.html [hunk 2]
```diff
@@ -17,7 +17,15 @@
                         <td data-title="Details">{{ test.test_type.description }} <a href="{{ test.github_link }}" target="_blank">{{ (test.commit[:7] if test.test_type == TestType.commit else test.pr_nr) }}</a></td>
                         <td data-title="Platform">{{ test.platform.description }}</td>
                         <td data-title="Run Errors">{{ "Unknown" if not test.finished else ("Yes" if test.failed else "No") }}</td>
-                        <td data-title="Completed"><a href="{{ url_for('test.by_id', test_id=test.id) }}">{{ "Yes" if test.finished else "No" }}</a></td>
+                        <td data-title="Completed">
+                            <a href="{{ url_for('test.by_id', test_id=test.id) }}">{{ "Yes" if test.finished else "No" }}</a>
+                            {% if user.is_admin or request.endpoint == 'custom.index' %}
```
Comment (canihavesomecoffee) [inline templates/test/list_table.html]: So any user that manager to get this view rendered from the correct place can restart the test... doesn't seem good.

Comment (satyammittal) [reply templates/test/list_table.html]: There is a check at backend too, which will check whether the test has been started by the same user or not.
Or other thing we can do is make a function call in test model, which will confirm whether to display start/stop button or not

Comment (canihavesomecoffee) [reply templates/test/list_table.html]: Even if there's a back-end check on the restart_test controller, we still want to hide it here whenever it's not applicable. A function in the test model isn't the correct place. Rather call this view with an extra variable that decides whether or not to show it.

**Pull request merged**